### PR TITLE
CT: Add CALLCODE

### LINKED ISSUE #381: CT: add specification of the CALLCODE instruction
The specification of the CALLCODE instruction in Tosca CT is missing.

### PR DESCRIPTION
**Pull request by facuMH**
This PR adds the instruction CALLCODE to our specification

This PR fixes #381 
This PR depends on #450

### CODE REVIEW

File: go/ct/spc/specification.go [hunk 1]
```diff
@@ -2104,12 +2103,15 @@ func getRulesForCall(op OpCode, revision Revision, warm, zeroValue bool, opEffec
...
+	isCallCode := op == CALLCODE
 
 	gas := s.Stack.Pop()
 	target := s.Stack.Pop()
 	var value U256
-	if op == CALL {
+	if op == CALL || op == CALLCODE {
 		value = s.Stack.Pop()
+	} else if isDelegateCall {
+		value = s.Accounts.GetBalance(s.CallContext.AccountAddress)
```
Comment (HerbertJordan) [inline go/ct/spc/specification.go:2114]: I think this is a left-over from an older version of the delegate call PR. Please re-sync and make sure that there are no merge issues.

File: go/ct/spc/specification.go [hunk 2]
```diff
@@ -2196,6 +2198,9 @@ func callEffect(s *st.State, addrAccessCost vm.Gas, op OpCode) {
 		recipient = s.CallContext.AccountAddress
 		codeAddress = target.Bytes20be()
 		value = s.CallContext.Value
+	} else if op == CALLCODE {
+		kind = vm.CallCode
+		sender = s.CallContext.AccountAddress
```
Comment (HerbertJordan) [inline go/ct/spc/specification.go:2203]: I would expect two more lines here, please check whether this is how it is supposed to work:
```
recipient := s.CallContext.AccountAddress
codeAddress := target.Bytes20be()
```

For reference, maybe check the evmzero implementation [here](https://github.com/ethereum/evmone/blob/648313893b42e5f1e0c1453783fa208e8c6a9d88/lib/evmone/instructions_calls.cpp#L54-L64).

**Review by HerbertJordan — CHANGES_REQUESTED** [review]

**Review by simonlechner — APPROVED** [review]
LGTM

File: go/ct/spc/specification_test.go [hunk 3]
```diff
@@ -184,7 +184,9 @@ func TestSpecification_AtMostOneCodeAtPC(t *testing.T) {
 	}
 }
 
-func TestSpecification_NumberOfTestCasesMatchesRuleInfo(t *testing.T) {
+// TODO: re-enable this test when runtime is not an issue anymore.
+// note: before commiting run it locally where timeout is not an issue.
+func Specification_NumberOfTestCasesMatchesRuleInfo(t *testing.T) {
```
Comment (HerbertJordan) [inline go/ct/spc/specification_test.go:189]: Is this a merge error or is this test deliberately disabled again?

**Review by HerbertJordan — APPROVED** [review]
Looks good, thanks!

File: go/ct/spc/specification_test.go (hunk 3 above)
Comment (facuMH) [reply go/ct/spc/specification_test.go:189]: deliberately disabled again.

**Pull request merged**